# Matrix Operations

In [2]:
# import libraries
import pandas as pd
import numpy as np

**What are Matrices?**

In simple terms, matrix is a two-dimensional array of numbers arranged in rows and columns.

Matrices are used to represent datasets, where each row corresponds to an observation or sample, and each column represents a feature or attribute of that sample.

In this project, I'll be using a movie dataset where each row represents a movie, and each column represents a feature.


In [5]:
# load data into dataframe
df = pd.read_csv('../datasets/Hydra-Movie-Scrape.csv')


In [8]:
# create movie feature matrix
movie_matrix = df[['Runtime', 'Rating']].to_numpy()

Now that we have our feature matrix, let's check its shape

In [9]:
movie_matrix.shape

(3886, 2)

It is a 3886 x 2 matrix, meaning it has 3886 rows and 2 columns. Our matrix is tall and skinny.

In [12]:
# extract movie titles
movie_titles = df['Title']


Let's now check out the Transpose of the matrix.